### PR TITLE
Drop support of Node 12 & 14

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue
There is no even security support for Node 12 now and 14 will meet end of life in 6 months https://endoflife.date/nodejs 

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/
## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: .github/workflows/osrm-backend.yml [hunk 1]
```diff
@@ -449,53 +463,6 @@ jobs:
             ENABLE_CONAN: ON
             NODE_PACKAGE_TESTS_ONLY: ON
 
-          - name: conan-macos-x64-release-node-lts
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:452]: LTS is Node 18 at the moment, i.e. it seems there is no reason to build it separately as we explicitly build Node 18 version now.

Comment (SiarheiFedartsou) [reply .github/workflows/osrm-backend.yml:452]: Just for better understanding: node-latest is Node 19 now.

File: .github/workflows/osrm-backend.yml [hunk 2]
```diff
@@ -327,74 +341,74 @@ jobs:
...
+            node: 16
             runs-on: ubuntu-20.04
             BUILD_TYPE: Debug
             CCOMPILER: clang-6.0
             CXXCOMPILER: clang++-6.0
             ENABLE_CONAN: ON
             NODE_PACKAGE_TESTS_ONLY: ON
 
-
-          - name: node-16-conan-linux-release
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:383]: We removed Node 12 jobs, but added Node 18 instead(which we implicitly had before, but named it "LTS")

**Review by DennisOSRM — APPROVED** [review]
👍

**Pull request merged**